lib/kubernetes-deploy/resource_discovery.rb

```diff
@@ -10,7 +10,7 @@ def initialize(namespace:, context:, logger:, namespace_tags:)
     end
 
     def crds(sync_mediator)
-      sync_mediator.get_all(CustomResourceDefinition.kind).map do |r_def|
+      @crds ||= sync_mediator.get_all(CustomResourceDefinition.kind).map do |r_def|
```

Comment:
That's one shortcoming of the current approach, unfortunately. There's a few solutions available to us here:
- Add discovered CRD specs to the top of the priority list. This seems risk-free since they should have no external dependencies
- In the future, think about using a dependency graph to produce the priority list (e.g. a CloudSQL has something that says `iDependOn: cloudsqls.stable.shopify.io`. I'd say this is out of scope for this PR.

It's unclear to me whether this issue is handled right now, anyway. Not saying we shouldn't fix it here, just a note

Reply:
> That's one shortcoming of the current approach, unfortunately.

What is the exact behaviour though? It still falls back on a KubernetesResource with the "dunno what to do here" message, right?

I'm on the fence about introducing an annotation-driven dependency graph, but FWIW we did have someone request customized sequencing for another reason earlier this week. Regardless, I agree it doesn't need to be handled in this PR.